src/Microsoft.TestPlatform.PlatformAbstractions/net451/System/ProcessHelper.cs

```diff
@@ -15,21 +17,98 @@ namespace Microsoft.VisualStudio.TestPlatform.PlatformAbstractions;
+                throw new Win32Exception();
+
+            // If processMachine is IMAGE_FILE_MACHINE_UNKNOWN mean that we're not running using WOW64 x86 emulation.
+            // If nativeMachine is IMAGE_FILE_MACHINE_ARM64 mean that we're running on ARM64 architecture device.
+            if (processMachine == IMAGE_FILE_MACHINE_UNKNOWN && nativeMachine == IMAGE_FILE_MACHINE_ARM64)
+            {
+                // To distinguish between ARM64 and x64 emulated on ARM64 we check the PE header of the current running executable.
+                return IsArm64Executable(currentProcess.MainModule.FileName);
+            }
+        }
+        catch (Exception)
+        {
+            // We should have Win32Exception() or IOException() but we don't want to block and we'll fallback to emulated path.
+        }
+
+        return false;
+    }
+
+    private static bool IsArm64Executable(string path)
+    {
+        // This document specifies the structure of executable (image) files
+        // https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#general-concepts
+        using Stream fs = new FileStream(path, FileMode.Open, FileAccess.Read);
+        {
+            using BinaryReader reader = new(fs);
+
+            // https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#ms-dos-stub-image-only
+            // At location 0x3c, the stub has the file offset to the PE signature. 
+            fs.Position = 0x3C;
+            var peHeader = reader.ReadUInt32();
+
+            // Check if the offset is invalid
+            if (peHeader > fs.Length - 5)
+            {
+                return false;
+            }
+
+            // https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#signature-image-only
+            // Moving to the PE Header start location.
+            fs.Position = peHeader;
+
+            // After the MS-DOS stub, at the file offset specified at offset 0x3c, is a 4-byte signature that identifies the file as a PE format image file.
+            // This signature is "PE\0\0" (the letters "P" and "E" followed by two null bytes).
+            uint signature = reader.ReadUInt32();
+            if (signature != 0x00004550)
+            {
+                return false;
+            }
+
+            // https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#coff-file-header-object-and-image
+            // At the beginning of an object file, or immediately after the signature of an image file, is a standard COFF file header.
+            var machine = reader.ReadUInt16();
+            reader.ReadUInt16(); //NumberOfSections
+            reader.ReadUInt32(); //TimeDateStamp
+            reader.ReadUInt32(); //PointerToSymbolTable
+            reader.ReadUInt32(); //NumberOfSymbols
+            reader.ReadUInt16(); //SizeOfOptionalHeader
+            reader.ReadUInt16(); //Characteristics
+
+            // https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#optional-header-image-only
```

Comment:
Same comment about culture

```suggestion
            // https://docs.microsoft.com/windows/win32/debug/pe-format#optional-header-image-only
```

Reply:
done